openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -357,9 +357,19 @@ def starts_within(self, days):
         """
         Returns True if the course starts with-in given number of days otherwise returns False.
         """
-
         return course_metadata_utils.course_starts_within(self.start, days)
 
+    def retrieve_datetime(self, get_start=True):
+        """
+        Returns the desired datetime string corresponding to the course's salient date
+        formatted as an ISO string.
+        """
+        if get_start is True:
+            return_date = self.start
+        else:
+            return_date = self.end
+        return return_date.strftime('%Y-%m-%dT%H:%M:%S%z')
+
```

Comment:
Unfortunately, this change is not that simple.  Looking at the methods that you are replacing below, those methods handle cases where the course team has entered free-style strings for the course dates.  So it won't always be a timestamp.  For example, it can be "Fall 2016" or "Halloween of 2017".